client/blocks/reader-site-subscription/subscribe-to-newsletter-categories.tsx

```diff
@@ -0,0 +1,88 @@
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import {
+	useNewsletterCategoriesQuery,
+	useNewsletterCategorySubscriptionMutation,
+	useSubscribedNewsletterCategories,
+} from 'calypso/data/newsletter-categories';
+import { NewsletterCategory } from 'calypso/data/newsletter-categories/types';
+
+type SubscribeToNewsletterCategoriesProps = {
+	siteId: number;
+};
+
+const convertToMutationFormat = (
+	allCategories: NewsletterCategory[],
+	subscribedCategoryIds: number[]
+) =>
+	allCategories.map( ( category ) => ( {
+		term_id: category.id,
+		subscribe: subscribedCategoryIds.includes( category.id ),
+	} ) );
+
+const SubscribeToNewsletterCategories = ( { siteId }: SubscribeToNewsletterCategoriesProps ) => {
+	const translate = useTranslate();
+	const { data: newsletterCategoriesData } = useNewsletterCategoriesQuery( { siteId } );
+	const { data: subscribedNewsletterCategoriesData } = useSubscribedNewsletterCategories( {
+		siteId,
+	} );
+	const subscribedCategoryIds = useMemo(
```

Comment:
You could get away here with only sending the category that's changed, you don't need to send them all. It's a tiny performance optimization that could be done.

Reply:
Nice! I'll do it then, thank you!